Lib/inspect.py

```diff
@@ -792,25 +792,42 @@ def findsource(object):
         return lines, 0
 
     if isclass(object):
-        name = object.__name__
-        pat = re.compile(r'^(\s*)class\s*' + name + r'\b')
-        # make some effort to find the best matching class definition:
-        # use the one with the least indentation, which is the one
-        # that's most probably not inside a function definition.
-        candidates = []
-        for i in range(len(lines)):
-            match = pat.match(lines[i])
-            if match:
-                # if it's at toplevel, it's already the best one
-                if lines[i][0] == 'c':
-                    return lines, i
-                # else add whitespace to candidate list
-                candidates.append((match.group(1), i))
-        if candidates:
-            # this will sort by whitespace, and by line number,
-            # less whitespace first
-            candidates.sort()
-            return lines, candidates[0][1]
+        import ast
+
+        class ClassVisitor(ast.NodeVisitor):
+
+            def __init__(self, qualname):
+                self.stack = []
+                self.line_number = None
+                self.qualname = qualname
+
+            def visit_ClassDef(self, node):
+                self.stack.append(node.name)
+
```

Comment:
I think it would be clearer if remove empty lines in this method. Left empty lines only between methods and classes.

Reply:
Removed empty lines. Thanks.